src/scimlfunctions.jl

```diff
@@ -3751,7 +3751,11 @@ function BVPFunction{iip, specialize, twopoint}(f, bc;
     end
 
     bciip = if !twopoint
-        isinplace(bc, 4, "bc", iip)
+        try
+            isinplace(bc, 4, "bc", iip)
+        catch e
+            isinplace(bc, 5, "bc", iip)
+        end
```

Comment:
Don't know if it's the best way of fixing the issue(bcs are now in the form of ```bc(res, du, u, p, t)```)

Reply:
no try catch, do this by having a separate function type. It should be the same as the dynamic ODE function

Reply:
Agree, this would be a better solution

Reply:
Done